balancer.go

```diff
@@ -288,3 +288,82 @@ func murmur2(data []byte) uint32 {
 
 	return h
 }
+
+// StickyPartitioner is an Balancer implementation that the Message key is NOT used as part of the balancing strategy
+// in this balancer. Messages with the same key are not guaranteed to be sent to the same partition.
+// If a partition is specified in the Message, use it
+// Otherwise choose the sticky partition that changes when the batch is full.
+type StickyPartitioner struct {
+	spc stickyPartitionCache
+}
+
+// Balance satisfies the Balancer interface.
+func (sp *StickyPartitioner) Balance(msg Message, partitions ...int) int {
+	return sp.balance(msg, partitions)
+}
+
+func (sp *StickyPartitioner) balance(msg Message, partitions []int) int {
+	return sp.spc.partition(msg, partitions)
+}
+
+// OnNewBatch changes the sticky partition If a batch completed for the current sticky partition.
+// Alternately, if no sticky partition has been determined, set one.
+func (sp *StickyPartitioner) OnNewBatch(msg Message, partitions []int, prevPartition int) int {
```

Comment:
it looks like sticky partitioning should keep track of a batch size per topic, and upon filling the batch, change to a new partition

It looks like here you intend to have the user of the partitioner change the partition themselves? Thoughts on keeping the batch size counter internally instead?